apiserver/facades/agent/uniter/goal-state_test.go

```diff
@@ -0,0 +1,325 @@
+	s.authorizer = apiservertesting.FakeAuthorizer{
+		Tag: s.mysqlUnit.Tag(),
+	}
+
+	// Create the resource registry separately to track invocations to
+	// Register.
+	s.resources = common.NewResources()
+	s.AddCleanup(func(_ *gc.C) { s.resources.StopAll() })
+
+	uniterAPI, err := uniter.NewUniterAPI(
+		s.State,
+		s.resources,
+		s.authorizer,
+	)
+	c.Assert(err, jc.ErrorIsNil)
+	s.uniter = uniterAPI
+}
+
+var (
+	timestamp          = time.Date(2200, time.November, 5, 0, 0, 0, 0, time.UTC)
+	charmStatusWaiting = params.GoalStateStatus{
+		Status: "waiting",
+		Since:  &timestamp,
+	}
+	expectedUnitWordpress = params.UnitsGoalState{
+		"wordpress/0": charmStatusWaiting,
+	}
+	expectedUnitMysql = params.UnitsGoalState{
+		"mysql/0": charmStatusWaiting,
+	}
+	expected2UnitsMysql = params.UnitsGoalState{
+		"mysql/0": charmStatusWaiting,
+		"mysql/1": charmStatusWaiting,
+	}
+)
+
+// TestGoalStatesNoRelation tests single application with single unit.
+func (s *uniterGoalStateSuite) TestGoalStatesNoRelation(c *gc.C) {
+	args := params.Entities{Entities: []params.Entity{
+		{Tag: "unit-mysql-0"},
+		{Tag: "unit-wordpress-0"},
+	}}
+	expected := params.GoalStateResults{
+		Results: []params.GoalStateResult{
+			{
+				Result: &params.GoalState{
+					Units: expectedUnitMysql,
+				},
+			}, {
+				Error: apiservertesting.ErrUnauthorized,
+			},
+		},
+	}
+	testGoalStates(c, s.uniter, args, expected)
+}
+
+// TestGoalStatesNoRelationTwoUnits adds a new unit to mysql application.
+func (s *uniterGoalStateSuite) TestPeerUnitsNoRelation(c *gc.C) {
+	args := params.Entities{Entities: []params.Entity{
+		{Tag: "unit-mysql-0"},
+		{Tag: "unit-wordpress-0"},
+	}}
+
+	addApplicationUnitToMachine(c, s.mysql, s.machine1)
+
+	expected := params.GoalStateResults{
+		Results: []params.GoalStateResult{
+			{
+				Result: &params.GoalState{
+					Units: expected2UnitsMysql,
+				},
+			}, {
+				Error: apiservertesting.ErrUnauthorized,
+			},
+		},
+	}
+	testGoalStates(c, s.uniter, args, expected)
+}
+
+// TestGoalStatesSingleRelation tests structure with two different
+// application units and one relation between the units.
+func (s *uniterGoalStateSuite) TestGoalStatesSingleRelation(c *gc.C) {
+
+	err := s.addRelationToSuiteScope(c, s.wordpressUnit, s.mysqlUnit)
+	c.Assert(err, jc.ErrorIsNil)
+
+	args := params.Entities{Entities: []params.Entity{
+		{Tag: "unit-mysql-0"},
+		{Tag: "unit-wordpress-0"},
+	}}
+	expected := params.GoalStateResults{
+		Results: []params.GoalStateResult{
+			{
+				Result: &params.GoalState{
+					Units: expectedUnitMysql,
+					Relations: map[string]params.UnitsGoalState{
+						"db":     expectedUnitWordpress,
+						"server": expectedUnitMysql,
+					},
+				},
+			}, {
+				Error: apiservertesting.ErrUnauthorized,
+			},
+		},
+	}
+	testGoalStates(c, s.uniter, args, expected)
+}
+
+// TestGoalStatesMultipleRelations tests GoalStates with three
+// applications one application has two units, and each unit is related
```

Comment:
Sorry, i should have picked this up before.
Units aren't related to units.
Relations are between applications. You can just delete the bit after the comma I think.